app/models/periodic_expense_runner.rb

```diff
@@ -0,0 +1,28 @@
+class PeriodicExpenseRunner
+  attr_accessor :today
+
+  def initialize
+    @today = Date.today
+  end
+
+  def run
+    periodic_expenses.each do |periodic_expense|
+      create_expense(periodic_expense)
+    end
+  end
+
+  private
+
+  def periodic_expenses
+    PeriodicExpense.where('start_date <= :today AND (end_date IS NULL OR end_date > :today)', today: today)
+  end
+
+  def create_expense(periodic_expense)
+    return unless periodic_expense.ready_to_pay?
+
+    name = "#{periodic_expense.name} (#{periodic_expense.period.downcase})"
+    periodic_expense.last_paid_at = today
+    expense = Expense.new(name: name, user: periodic_expense.user, date: today.to_datetime, value: periodic_expense.amount)
+    expense.save
```

Comment:
1. You can just do `Expense.create` instead of calling `save right after`
2. Are you updating the periodic_expense at all? You're setting the new last_paid_at, but I don't see the model being saved anywhere
3. Assuming the save is missing, then both saves should be wrapped around a transaction (http://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html), so that it becomes an atomic operation